src/main/java/com/pusher/client/channel/impl/PresenceChannelImpl.java

```diff
@@ -65,35 +66,10 @@ else if (event.equals(MEMBER_REMOVED_EVENT)) {
     }
 
     @Override
-    @SuppressWarnings("rawtypes")
     public String toSubscribeMessage() {
-
-        final String authResponse = getAuthResponse();
-
-        try {
-            final Map authResponseMap = GSON.fromJson(authResponse, Map.class);
-            final String authKey = (String)authResponseMap.get("auth");
-            final Object channelData = authResponseMap.get("channel_data");
-
-            storeMyUserId(channelData);
-
-            final Map<Object, Object> jsonObject = new LinkedHashMap<Object, Object>();
-            jsonObject.put("event", "pusher:subscribe");
-
-            final Map<Object, Object> dataMap = new LinkedHashMap<Object, Object>();
-            dataMap.put("channel", name);
-            dataMap.put("auth", authKey);
-            dataMap.put("channel_data", channelData);
-
-            jsonObject.put("data", dataMap);
-
-            final String json = GSON.toJson(jsonObject);
-
-            return json;
-        }
-        catch (final Exception e) {
-            throw new AuthorizationFailureException("Unable to parse response from Authorizer: " + authResponse, e);
-        }
+        String msg = super.toSubscribeMessage();
+        myUserID = extractUserIdFromChannelData((String)channelData);
```

Comment:
```suggestion
        myUserID = extractUserIdFromChannelData(channelData);
```
I don't think the cast is necessary.